### PR TITLE
ddl(ticdc): support batch create table ddl

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/11009

#### Tests 

 - Integration test
 - Manual test (add detailed scripts or steps below)
1. create changefeed.
2. set global tidb_enable_fast_create_table=on; for the upstream tidb'
3. concurrently create 1000 tables in upstream, and insert data
4. check the tables and data in the downstream tidb
image

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by sdojjy — APPROVED** [review]

File: tests/integration_tests/batch_add_table/run.sh [hunk 1]
```diff
@@ -39,7 +36,20 @@ function run_with_fast_create_table() {
 	pulsar) run_pulsar_consumer --upstream-uri $SINK_URI ;;
 	esac
 
+	## to generate batch create ddl
```
Comment (CharlesCheung96) [inline tests/integration_tests/batch_add_table/run.sh:39]: please test the filter

File: cdc/puller/ddl_puller.go [hunk 2]
```diff
@@ -390,6 +391,31 @@ func (p *ddlJobPullerImpl) handleJob(job *timodel.Job) (skip bool, err error) {
...
+		skip = true
+		for index, tableInfo := range multiTableInfos {
+			// judge each table whether need to be skip
+			if p.filter.ShouldDiscardDDL(job.Type, job.SchemaName, tableInfo.Name.O) {
+				continue
+			}
+			newMultiTableInfos = append(newMultiTableInfos, multiTableInfos[index])
+			newQuerys = append(newQuerys, querys[index])
+			skip = false
+		}
```
Comment (CharlesCheung96) [inline cdc/puller/ddl_puller.go:412]:
```suggestion
		}
		skip = len(newMultiTableInfos) == 0
```

File: cdc/puller/ddl_puller.go [hunk 3]
```diff
@@ -390,6 +391,31 @@ func (p *ddlJobPullerImpl) handleJob(job *timodel.Job) (skip bool, err error) {
...
+
+		skip = true
+		for index, tableInfo := range multiTableInfos {
+			// judge each table whether need to be skip
+			if p.filter.ShouldDiscardDDL(job.Type, job.SchemaName, tableInfo.Name.O) {
+				continue
+			}
+			newMultiTableInfos = append(newMultiTableInfos, multiTableInfos[index])
+			newQuerys = append(newQuerys, querys[index])
+			skip = false
```
Comment (CharlesCheung96) [inline cdc/puller/ddl_puller.go:411]:
```suggestion
```

File: cdc/puller/ddl_puller.go [hunk 4]
```diff
@@ -390,6 +391,31 @@ func (p *ddlJobPullerImpl) handleJob(job *timodel.Job) (skip bool, err error) {
...
+	case timodel.ActionCreateTables:
+		// we only use multiTableInfos and Querys when we generate job event
+		// So if some table should be discard, we just need to delete the info from multiTableInfos and Querys
+		var newMultiTableInfos []*timodel.TableInfo
+		var newQuerys []string
+
+		multiTableInfos := job.BinlogInfo.MultipleTableInfos
+		querys := strings.Split(job.Query, ";")
+
+		skip = true
```
Comment (CharlesCheung96) [inline cdc/puller/ddl_puller.go:403]:
```suggestion
```

File: cdc/puller/ddl_puller.go [hunk 5]
```diff
@@ -390,6 +391,31 @@ func (p *ddlJobPullerImpl) handleJob(job *timodel.Job) (skip bool, err error) {
...
+			// judge each table whether need to be skip
+			if p.filter.ShouldDiscardDDL(job.Type, job.SchemaName, tableInfo.Name.O) {
+				continue
+			}
+			newMultiTableInfos = append(newMultiTableInfos, multiTableInfos[index])
+			newQuerys = append(newQuerys, querys[index])
+			skip = false
+		}
+		// if some ddls are discarded
+		if len(newQuerys) != len(querys) {
```
Comment (CharlesCheung96) [inline cdc/puller/ddl_puller.go:414]:
```suggestion
```
remove dummy check

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**